yarn-project/validator-client/src/validator.ts

```diff
@@ -75,13 +75,18 @@ export class ValidatorClient implements Validator {
   // Target is temporarily hardcoded, for a test, but will be calculated from smart contract
   // TODO(https://github.com/AztecProtocol/aztec-packages/issues/7962)
   // TODO(https://github.com/AztecProtocol/aztec-packages/issues/7976): require suitable timeouts
-  async collectAttestations(slot: bigint, numberOfRequiredAttestations: number): Promise<BlockAttestation[]> {
+  async collectAttestations(
+    proposal: BlockProposal,
+    numberOfRequiredAttestations: number,
+  ): Promise<BlockAttestation[]> {
     // Wait and poll the p2pClients attestation pool for this block
     // until we have enough attestations
 
-    this.log.info(`Waiting for attestations for slot, ${slot}`);
+    const slot = proposal.header.globalVariables.slotNumber.toBigInt();
 
-    let attestations: BlockAttestation[] = [];
+    this.log.info(`Waiting for ${numberOfRequiredAttestations} attestations for slot: ${slot}`);
+
+    let attestations: BlockAttestation[] = [await this.attestToProposal(proposal)];
```

Comment:
I always attest to my own proposal?

Reply:
The smart contract does not filter the proposer from the expected signatures, this could probably be removed by doing the filtering